demisto_sdk/commands/download/downloader.py

```diff
@@ -183,6 +235,83 @@ def fetch_custom_content(self) -> bool:
+
+        return endpoint, req_type, req_body
+
+    def get_system_automation(self, req_type):
+        automation_list: list = []
+        for script in self.input_files:
+            endpoint = f"automation/load/{script}"
+            api_response = demisto_client.generic_request_func(self.client, endpoint, req_type)
+            automation_list.append(ast.literal_eval(api_response[0]))
+        return automation_list
+
+    def arrange_response(self, system_items_list):
+        if self.system_item_type == 'Playbook':
+            system_items_list = system_items_list.get('playbooks')
+
+        if self.system_item_type in ['Classifier', 'Mapper']:
+            system_items_list = system_items_list.get('classifiers')
+
+        return system_items_list
+
+    def build_file_name(self, item):
+        item_name: str = item.get('name') or item.get('id')
+        return item_name.replace('/', ' ').replace(' ', '_') + ITEM_TYPE_TO_PREFIX[self.system_item_type]
+
+    def fetch_system_content(self):
+        """
+        Fetches the system content from Demisto into a temporary dir.
+        :return: True if fetched successfully, False otherwise
+        """
+        try:
+            system_items_list: Union
+            endpoint, req_type, req_body = self.build_req_params()
+
+            if self.system_item_type == 'Automation':
+                system_items_list = self.get_system_automation(req_type)
+
+            else:
+                api_response = demisto_client.generic_request_func(self.client, endpoint, req_type, body=req_body)
+                system_items_list = ast.literal_eval(api_response[0])
+
+            self.arrange_response(system_items_list)
+
+            for item in system_items_list:  # type: ignore
+                file_name: str = self.build_file_name(item)
+                file_path: str = os.path.join(self.system_content_temp_dir, file_name)
+                with open(file_path, 'w') as file:
+                    if file_path.endswith('json'):
+                        json.dump(item, file)
+                    else:
+                        yaml.dump(item, file)
+            return True
+
+        except ApiException as e:
+            if e.status == 401:
+                print_color('\nVerify that the environment variable DEMISTO_API_KEY is configured properly.\n',
+                            LOG_COLORS.RED)
+            print_color(f'Exception raised when fetching custom content:\nStatus: {e}', LOG_COLORS.NATIVE)
+            return False
+        except MaxRetryError as e:
+            print_color('\nVerify that the environment variable DEMISTO_BASE_URL is configured properly.\n',
+                        LOG_COLORS.RED)
+            print_color(f'Exception raised when fetching custom content:\n{e}', LOG_COLORS.NATIVE)
+            return False
+        except Exception as e:
+            print_color(f'Exception raised when fetching custom content:\n{e}', LOG_COLORS.NATIVE)
```

Comment:
Is this logic copied from anywhere or brand new?

Reply:
done